src/components/store/CartOptionsGroup/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { Dropdown, Typography } from '@/components/common';
+import { useId } from 'react';
+import styles from './style.module.scss';
+
+interface CartOptionGroupProps {
+  currOption: string | undefined;
+  inStock: boolean;
+  inCart: boolean;
+  lifetimeRemaining: number;
+  monthlyRemaining: number;
+  amountToBuy: number;
+  optionsKey: string;
+  onCartChange: (inCart: boolean) => void;
+  onAmountChange: (amountToBuy: number) => void;
+}
+
+const CartOptionsGroup = ({
+  currOption,
+  inStock,
+  inCart,
+  onCartChange,
+  lifetimeRemaining,
+  monthlyRemaining,
+  amountToBuy,
+  optionsKey,
+  onAmountChange,
+}: CartOptionGroupProps) => {
+  const myID = useId();
+
+  const maxCanBuy = Math.min(20, Math.min(lifetimeRemaining, monthlyRemaining));
+
+  let buyButtonText = 'Add to Cart';
+
+  if (maxCanBuy === 0) {
+    buyButtonText = 'Limit Reached';
+  } else if (currOption === undefined && optionsKey !== undefined) {
+    buyButtonText = `Select a ${optionsKey}`;
+  } else if (inStock) {
+    buyButtonText = 'Add to Cart';
+  } else {
+    buyButtonText = 'Out of Stock';
+  }
+
+  const optionArr: Array<{ value: string; label: string }> = [];
+
+  for (let i = 1; i <= maxCanBuy; i += 1) {
+    optionArr.push({ value: `${i}`, label: `${i}` });
+  }
```

Comment:
nit: I think 
```tsx
Array(maxCanBuy).fill().map((_, i) => i+1);
```
would be more efficient? Can add a comment explaining what this is doing also